src/bin/configurations/mod.rs

```diff
@@ -0,0 +1,483 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::{parser::ValueSource, ArgMatches};
+use solang::{
+    codegen::{OptimizationLevel, Options},
+    file_resolver::FileResolver,
+    Target,
+};
+
+use std::{collections::HashSet, ffi::OsString, fs, path::PathBuf, process::exit};
+use toml::Table;
+mod unit_tests;
+
+// A macro for error handling.
+macro_rules! match_err {
```

Comment:
Is this macro necessary? I think instead of invoking this macro in most places you could just do `.map_err(ToString::to_string)?`, or, in case the error does not implement `ToString` but `Display`, just directly call `.map_err(|e| format!("{e}"))?`.